### PR TITLE
Geode 7850: Implement Redis Rename Command

### PR DESCRIPTION
**Pull request by xtreme-prasath**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by prettyClouds** [comment]
@jdeppe-pivotal @kohlmu-pivotal can you two take a look.  We can't seem to add reviewers via github.

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/RenameTest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  private static ThreePhraseGenerator generator = new ThreePhraseGenerator();
+
+  @BeforeClass
+  public static void setUp() {
+    rand = new Random();
+    CacheFactory cf = new CacheFactory();
+    cf.set(LOG_LEVEL, "info");
+    cf.set(MCAST_PORT, "0");
+    cf.set(LOCATORS, "");
+    cache = cf.create();
```
Comment (kohlmu-pivotal) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/RenameTest.java]: Please don't use single (or double) character variable names

File: geode-redis/src/main/java/org/apache/geode/redis/internal/ExecutionHandlerContext.java [hunk 2]
```diff
@@ -101,13 +100,10 @@ public PubSub getPubSub() {
    * @param pwd Authentication password for each context, can be null
    */
   public ExecutionHandlerContext(Channel ch, Cache cache, RegionProvider regionProvider,
-      GeodeRedisServer server, byte[] pwd, KeyRegistrar keyRegistrar,
-      PubSub pubSub,
-      RedisLockService hashLockService,
-      RedisLockService setLockService) {
+      GeodeRedisServer server, byte[] pwd, KeyRegistrar keyRegistrar, PubSub pubSub,
```
Comment (kohlmu-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/ExecutionHandlerContext.java]: what is `pwd`?

File: geode-redis/src/main/java/org/apache/geode/redis/internal/ExecutionHandlerContext.java [hunk 3]
```diff
@@ -101,13 +100,10 @@ public PubSub getPubSub() {
...
+      RedisLockService lockService) {
     this.keyRegistrar = keyRegistrar;
-    this.hashLockService = hashLockService;
-    this.setLockService = setLockService;
+    this.lockService = lockService;
     this.pubSub = pubSub;
     if (ch == null || cache == null || regionProvider == null || server == null) {
```
Comment (kohlmu-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/ExecutionHandlerContext.java]: `ch` is bad parameter/variable name. Please correct to something a little more descriptive. Like `channel`... makes it easier to read when one gets past 15 lines of code where it was initially defined

**Comment by prettyClouds** [comment]
> The code is somehow lacking in logging... Makes it REALLY hard to understand what is happening on the server WITHOUT stepping through code. Could we please add logging to `RenameExecutor`. Things like logger.info, logger.debug, logger.warn and logger.error are incredibly useful.

@kohlmu-pivotal, the redis commands all execute within this `ExecutionHandlerContext`.  The `ExecutionHandlerContext` logs all the inputs and then will also log any exceptions thrown.  I think that should make the `RenameExecutor` pretty debuggable

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/RenameTest.java (hunk 1 above)
Comment (prettyClouds) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/RenameTest.java]: done. thx

File: geode-redis/src/main/java/org/apache/geode/redis/internal/ExecutionHandlerContext.java (hunk 2 above)
Comment (prettyClouds) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/ExecutionHandlerContext.java]: done. thx

File: geode-redis/src/main/java/org/apache/geode/redis/internal/ExecutionHandlerContext.java (hunk 3 above)
Comment (prettyClouds) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/ExecutionHandlerContext.java]: done. thx

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/GetExecutor.java [hunk 4]
```diff
@@ -36,6 +35,8 @@ public void executeCommand(Command command, ExecutionHandlerContext context) {
 
     ByteArrayWrapper key = command.getKey();
     checkDataType(key, RedisDataType.REDIS_STRING, context);
+
+    Region<ByteArrayWrapper, ByteArrayWrapper> r = context.getRegionProvider().getStringsRegion();
```
Comment (kohlmu-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/GetExecutor.java]: r?

**Pull request merged**